eth/api.go

```diff
@@ -281,10 +282,21 @@ func NewPublicDebugAPI(eth *Ethereum) *PublicDebugAPI {
 }
 
 // DumpBlock retrieves the entire state of the database at a given block.
-func (api *PublicDebugAPI) DumpBlock(number uint64) (state.Dump, error) {
-	block := api.eth.BlockChain().GetBlockByNumber(number)
+func (api *PublicDebugAPI) DumpBlock(blockNr rpc.BlockNumber) (state.Dump, error) {
+
+	var block *types.Block
+
+	switch blockNr {
+	case rpc.PendingBlockNumber:
+		// Pending block is only known by the miner
+		block = api.eth.miner.PendingBlock()
```

Comment:
This code path will need a bit more special handling. The "dumping" code below in line 301 looks up the state from the real blockchain. If we're dumping the pending state, you actually need to request both the pending block as well as the pending state from the miner and operate on those versus the current approach.